### PR TITLE
Fix schema parse error when using reserved keys as table names

### LINKED ISSUE #874: Unable to Parse Schema
I am receiving the following error on the latest version of 

```
Microsoft.Azure.WebJobs.Extensions.Sql: Encountered error(s) while parsing schema and object name:
Incorrect syntax near ..
```

Here is one of my functions:
```cs


    [FunctionName("User_SqlTrigger")]
    public async Task Run(
        [SqlTrigger("[dbo].[User]",  "DefaultConnection")]
        IReadOnlyList<SqlChange<UserTriggerRow>> changes,
        ILogger logger)
    {
    
    // ..
    }
```

Here are my packages:
```xml
  <ItemGroup>
        <PackageReference Include="Azure.Extensions.AspNetCore.Configuration.Secrets" Version="1.2.2" />
        <PackageReference Include="Azure.Security.KeyVault.Secrets" Version="4.5.0" />
        <PackageReference Include="Microsoft.ApplicationInsights.WorkerService" Version="2.21.0" />
        <PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.1.0" />
        <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Http" Version="3.2.0" />
        <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.ServiceBus" Version="5.10.0" />

        <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Sql" Version="3.0.253-preview" />
        <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.16" />
        <PackageReference Include="Microsoft.Extensions.Configuration.UserSecrets" Version="6.0.1" />
        <PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
        <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="4.1.3" />
        <PackageReference Include="RestSharp" Version="109.0.1" />
    </ItemGroup>
```

### PR DESCRIPTION
**Pull request by MaddyDev**
Fixes #874 

When using tables with reserved key names Ex: User, Into, etc. we have to use the bracketed value for sql executions to be successful. We've been passing bracketed values in most places but have missed in few which resulted in the exception. Updated those in this PR and added a test.

### CODE REVIEW

File: test/Integration/SqlTriggerBindingIntegrationTests.cs [hunk 1]
```diff
@@ -656,6 +656,72 @@ public async void LastAccessTimeColumn_Created_OnStartup()
             Assert.True(1 == (int)this.ExecuteScalar("SELECT 1 FROM sys.columns WHERE Name = N'LastAccessTime' AND Object_ID = Object_ID(N'[az_func].[GlobalState]')"), $"{GlobalStateTableName} should have {LastAccessTimeColumnName} column after restarting the listener.");
         }
 
+        /// <summary>
+        /// Tests that trigger function executes on table whose name is a reserved word (User).
+        /// </summary>
+        [Theory]
+        [SqlInlineData()]
+        [UnsupportedLanguages(SupportedLanguages.Java)] // test timing out for Java
```
Comment (Charles-Gagnon) [inline test/Integration/SqlTriggerBindingIntegrationTests.cs:664]: Please open an issue to track fixing this

File: test/Integration/SqlTriggerBindingIntegrationTests.cs [hunk 2]
```diff
@@ -656,6 +656,72 @@ public async void LastAccessTimeColumn_Created_OnStartup()
...
+            int index = 0;
+            string messagePrefix = "SQL Changes: ";
+
+            var taskCompletion = new TaskCompletionSource<bool>();
+
+            void MonitorOutputData(object sender, DataReceivedEventArgs e)
+            {
+                if (e.Data != null && (index = e.Data.IndexOf(messagePrefix, StringComparison.Ordinal)) >= 0)
+                {
+                    string json = e.Data[(index + messagePrefix.Length)..];
```
Comment (Charles-Gagnon) [inline test/Integration/SqlTriggerBindingIntegrationTests.cs:680]: We have this logic in a couple of places now, could you make a issue to follow up on for consolidating this into a single helper?

File: src/TriggerBinding/SqlTriggerListener.cs [hunk 3]
```diff
@@ -82,8 +82,8 @@ public SqlTriggerListener(string connectionString, string tableName, string user
             }
             this._hasConfiguredMaxChangesPerWorker = configuredMaxChangesPerWorker != null;
 
-            this._scaleMonitor = new SqlTriggerScaleMonitor(this._userFunctionId, this._userTable.FullName, this._connectionString, this._maxChangesPerWorker, this._logger);
-            this._targetScaler = new SqlTriggerTargetScaler(this._userFunctionId, this._userTable.FullName, this._connectionString, this._maxChangesPerWorker, this._logger);
+            this._scaleMonitor = new SqlTriggerScaleMonitor(this._userFunctionId, this._userTable.BracketQuotedFullName, this._connectionString, this._maxChangesPerWorker, this._logger);
```
Comment (Charles-Gagnon) [inline src/TriggerBinding/SqlTriggerListener.cs:85]: I'd suggest passing in the entire table object - that way we're much more confident that the name is being escaped correctly (and don't accidently use the wrong name if we make changes there in the future)

File: test/Integration/SqlTriggerBindingIntegrationTests.cs (hunk 1 above)
Comment (MaddyDev) [reply test/Integration/SqlTriggerBindingIntegrationTests.cs:664]: #883

**Pull request merged**